### PR TITLE
refactor: vectorize matryoshka ce loss forward

### PR DESCRIPTION
**Pull request by leehanchung**

### CODE REVIEW

**Review by adityakusupati — APPROVED** [review]
LGTM!

File: MRL.py [hunk 1]
```diff
@@ -13,11 +13,11 @@ def __init__(self, relative_importance=None, **kwargs):
...
 	def forward(self, output, target):
-		loss=0
-		N = len(output)
-		for i in range(N):
-			rel = 1.0 if self.relative_importance is None else self.relative_importance[i] 
-			loss += rel*self.criterion(output[i], target)
+		if self.relative_importance is None:
+			loss = self.criterion(output, target).sum()
+		else:
+			weighted_losses = self.relative_importance * self.criterion(output, target)
```
Comment (adityakusupati) [inline MRL.py:19]: I am not sure if this will broadcast it right. 

https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

input shape is (N, C, d_1, d_2... d_k), output shape is (N, d_1, d_2... d_k),

G is nesting granularities (len(output)
In this case "output" is (G, C) ideally the for loop gets to output (G)

Your implementation gets something that I am not sure.

File: MRL.py [hunk 2]
```diff
@@ -13,11 +13,11 @@ def __init__(self, relative_importance=None, **kwargs):
 		self.relative_importance = relative_importance
 
 	def forward(self, output, target):
-		loss=0
-		N = len(output)
-		for i in range(N):
-			rel = 1.0 if self.relative_importance is None else self.relative_importance[i] 
-			loss += rel*self.criterion(output[i], target)
+		if self.relative_importance is None:
+			loss = self.criterion(output, target).sum()
```
Comment (adityakusupati) [inline MRL.py:17]: No need of .sum()

**Review by adityakusupati — COMMENTED** [review]
Any testing to show the broadcast and for loop are the same is good.

File: MRL.py (hunk 2 above)
Comment (adityakusupati) [reply MRL.py:17]: The PR is unmerged for now. @leehanchung
